src/main/java/org/sunbird/assessment/service/CohortsServiceImpl.java

```diff
@@ -303,14 +303,14 @@ private List<String> fetchBatchIdDetails(String contentId) {
 		return Collections.emptyList();
 	}
 
-	private List<SunbirdApiBatchResp> fetchBatchsDetails(String contentId) {
+	private List<SunbirdApiBatchResp> fetchBatchesDetails(String contentId) {
 		try {
-			// TODO - Call contentService.searchLiveContent(contentId)
-			// Get the batch details from
-			// SunbirdApiResp contentHierarchy = contentHierarchy.getResult().getContent()[0].getBatches();
-			SunbirdApiResp contentHierarchy = contentService.getHeirarchyResponse(contentId);
-			if (contentHierarchy != null && "successful".equalsIgnoreCase(contentHierarchy.getParams().getStatus())) {
-				return contentHierarchy.getResult().getContent().getBatches();
+			Map<String, Object> contentResponse = contentService.searchLiveContent(contentId);
+			Map<String, Object> contentResult = (Map<String, Object>) contentResponse.get(Constants.RESULT);
+			List<Map<String, Object>> content = (List<Map<String, Object>>) contentResult.get(Constants.CONTENT);
+			List<SunbirdApiBatchResp> contentHierarchy = (List<SunbirdApiBatchResp>) content.get(0).get(Constants.BATCHES);
```

Comment:
Need to check array length before accessing it..
is it automatically converting the Object to SunbirdApiBatchResp class ?